source/_posts/durable-entities.md

```diff
@@ -0,0 +1,129 @@
+
+## Using the Entity
+
+Now let's have a look at a simple client function that would make use of this `BedEntity`:
+
+```csharp
+public static class BedManager
+{
+    [FunctionName(nameof(AssignBed))]
+    public static async Task<IActionResult> AssignBed(
+        [HttpTrigger(AuthorizationLevel.Function, "get")]
+        HttpRequest req,
+        [DurableClient] IDurableEntityClient durableEntityClient,
+        ILogger log)
+    {
+        var bedNumber = req.Query["bedNumber"];
+        var entityId = new EntityId(nameof(BedEntity),bedNumber);
+        
+        log.LogInformation($"Received request for bed {bedNumber}");
+        
+        var bedEntity = await durableEntityClient.ReadEntityStateAsync<BedEntity>(entityId);
+        if (bedEntity.EntityExists && bedEntity.EntityState.IsOccupied)
+        {
+            return new BadRequestObjectResult("Bed is already occupied.");
+        }
+        
+        await durableEntityClient.SignalEntityAsync(entityId, nameof(BedEntity.AssignBedAsync));
+
+        return new OkObjectResult($"Bed {bedNumber} has been set to occupied.");
+    }
+}
+```
+
+If we now run this client Function and send a request to it (bedNumber = 123) with a breakpoint on the logline we see the entityId being constructed as `@BedEntity@123`. The framework will use this id to get the current state if an entity is found using this id. If an entity is found using the `ReadEntityStateAsync` method on the `IDurableEntityClient` we check if the bed is also occupied. If that is the case we cannot assign it and return a `BadRequest`. If no BedEntity could be found with this id or the bed is not occupied we signal the entity to assign a bed with the provided `bedNumber` using the `SignalEntityAsync` method. This method gets the entity and triggers the `AssignBedAsync` call which changes the state. The framework then persists the changes in the entity state back to table storage.
+
+But what happens in the underlaying framework if we call `SignalEntityAsync`? What happens is that the framework starts up a virtual actor as Microsoft calls it. Its a function that acts on our request to process information regarding the entities state. For the communication, the framework uses the existing durable queues that we know from the already existing durable functions framework which this is built on.
+
+Durable entities are in no way replacing the previously existing durable functions. They are an addition to the existing durable function framework to help you to get rid of some of the plumbing work that was still there. You can use all the parts that were already there in combination with this new addition. 
+
+![Durable Function Types](images/durable-entities/functiontypes.png)
+
```

Comment:
iets meer afsluitende conclusie, nu gooi je hier nog ff ene hele fijne tip neer.
je kan hierbij beter terug pakken op je client: van zie je dat we geen locks/checks hoeve te doen want dit wordt voor ons gedaan

Reply:
in je intro begin je vaak me een vraag stuk, en daar kun je in de conclusie juist op terug pakken:

'ofcourse this was a simple example, but i hope it might help you to get started with setting up small entity durable functions. just start with a few minor functions and work out from there'


zoiets??? 🤒